src/app/clusters/electrical-energy-measurement-server/ElectricalEnergyMeasurementCluster.cpp

```diff
@@ -16,237 +16,364 @@
  *    limitations under the License.
  */
 #include "ElectricalEnergyMeasurementCluster.h"
-#include "electrical-energy-measurement-server.h"
 
 #include <protocols/interaction_model/StatusCode.h>
 
-#include <app/AttributeAccessInterfaceRegistry.h>
-#include <app/AttributeValueEncoder.h>
-#include <app/EventLogging.h>
+#include <app/data-model/Decode.h>
+#include <app/data-model/Encode.h>
 #include <app/reporting/reporting.h>
-#include <app/util/attribute-storage.h>
-#include <app/util/config.h>
-#include <zap-generated/gen_config.h>
+#include <app/server-cluster/AttributeListBuilder.h>
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <clusters/ElectricalEnergyMeasurement/Attributes.h>
+#include <clusters/ElectricalEnergyMeasurement/Enums.h>
+#include <clusters/ElectricalEnergyMeasurement/Events.h>
+#include <clusters/ElectricalEnergyMeasurement/Metadata.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
 
 using chip::Protocols::InteractionModel::Status;
 
 namespace chip {
 namespace app {
 namespace Clusters {
-namespace ElectricalEnergyMeasurement {
 
-using namespace chip;
-using namespace chip::app::Clusters::ElectricalEnergyMeasurement::Attributes;
-using namespace chip::app::Clusters::ElectricalEnergyMeasurement::Structs;
+using namespace ElectricalEnergyMeasurement::Attributes;
+using namespace ElectricalEnergyMeasurement::Structs;
+using namespace ElectricalEnergyMeasurement;
 
-MeasurementData gMeasurements[MATTER_DM_ELECTRICAL_ENERGY_MEASUREMENT_CLUSTER_SERVER_ENDPOINT_COUNT +
-                              CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT];
+ElectricalEnergyMeasurementCluster::ElectricalEnergyMeasurementCluster(const Config & config) :
+    DefaultServerCluster({ config.endpointId, ElectricalEnergyMeasurement::Id }), mFeatureFlags(config.mFeatureFlags),
+    mEnabledOptionalAttributes(config.mEnabledOptionalAttributes)
+{
+    // Verify that optional attributes match the enabled features
+    // CumulativeEnergyImported requires kCumulativeEnergy + kImportedEnergy
+    VerifyOrDieWithMsg(!mEnabledOptionalAttributes.IsSet(CumulativeEnergyImported::Id) ||
+                           (mFeatureFlags.Has(Feature::kCumulativeEnergy) && mFeatureFlags.Has(Feature::kImportedEnergy)),
```

Comment:
This check is wrong.  The conformance here is `IMPE & CUME`, not `[IMPE & CUME]`.

Which means that you can in fact not even have an optional attribute flag for this and just derive whether this attribute is supported entirely from the feature map, no?

Reply:
Right, so with the config deriving the mEnabledOptionalAttributes, we do not need these tests at all.